Automattic/jetpack
# Simple Payments: CPTs and Shortcode

This is Jetpack part of architecture described in p8RSuw-2E-code #comment-51

- Introduces custom post type for orders and products (products being stored payment button configurations)
- Enables these custom post types over /posts endpoints
- describes [order data structure](https://github.com/Automattic/jetpack/pull/7409/files#diff-c544e8ee5ba304a9462be2f0365d2718R46)
- describes [product data structre](https://github.com/Automattic/jetpack/pull/7409/files#diff-c544e8ee5ba304a9462be2f0365d2718R88)
- introduces JS dependencies for displaying the checkout button
- introduces a shortcode to make checkout button appear

After this is merged in JP, we will port it to wpcom.
We will communicate with JP sites using the code below and with wpcom sites via reqular wp_query.

## Testing


- Check out this branch to your JP sandbox
- Create new product for your store using code below
(replace blog_id and domain with your JP sandbox)

```
function test_meta( $blog_id, $domain ) {
	$token = Jetpack_Data::get_access_token_by_blog_id_user_id( $blog_id, wpcom_get_blog_owner( $blog_id ) );
	$installed = Jetpack_Client::json_api_request( array(
	    'url' => 'https://public-api.wordpress.com/rest/v1.1/sites/'.$domain.'/posts/new',
	    'blog_id' => $blog_id,
	    'jetpack_token' => $token,
	    'method'=> 'POST',
	    'body' => json_encode(
	        array(
	        	'type' => 'jp_pay_product',
				'title'=>'Best potato in the whole universe',
				'content' => 'Buy our organic best potato. It is the best.',
	            'metadata' => array(
	                array(
	                	'key' => 'spay_price',
						'value' =>  '34.00'
	                ),
	                array(
	                	'key' => 'spay_currency',
	                	'value' => 'USD'
	                ),
	                array(
	                	'key' => 'spay_cta',
	                	'value' => 'Buy'
	                ),
	                array(
	                	'key' => 'spay_email',
	                	'value' => 'artur.piszek+test2@automattic.com'
	                ),
	                array(
	                	'key' => 'spay_multiple',
	                	'value' => '0'
	                ),
	                array(
	                	'key' => 'spay_status',
	                	'value' => 'enabled'
	                )
	            )
	        )
	    )
	) );

	print_r( $installed );	
}
test_meta( 118642378, 'arturpiszek.wpsandbox.me' );
```

- note the outputted postID
- go to your JP site, use the shortcode:

```
[simple-payment id=84]
```
(replace ID with your post id)

See the paypal button appear:
![zrzut ekranu 2017-07-03 o 17 13 44](https://user-images.githubusercontent.com/3775068/27798948-9a9e1690-6013-11e7-9605-99a5fd4a23f6.png)

- You can test a full flow if you sandbox D6278-code


CC @lamosty @sirbrillig @jhnstn @retrofox @jsnajdr
